samples/notes/android/src/main/kotlin/com/dropbox/market/notes/android/data/api/NotesApi.kt

```diff
@@ -0,0 +1,61 @@
+package com.dropbox.market.notes.android.data.api
+
+import com.dropbox.market.notes.android.Key
+import com.dropbox.market.notes.android.data.MarketNote
+import com.dropbox.market.notes.android.data.Notebook
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromJsonElement
+
+class NotesApi(private val client: HttpClient) : Api<Key, Notebook> {
+    override suspend fun get(key: Key): Notebook? {
+        return try {
+            when (key) {
+                Key.All -> {
+                    val response = client.get("$API_URL/notes")
+                    val serializer = Json { ignoreUnknownKeys = true }
+                    val body = serializer.decodeFromJsonElement<NotesApiResponse>(response.body())
+                    Notebook.Notes(body.value)
+                }
+
+                is Key.Single -> {
+                    val response = client.get("$API_URL/notes/${key.key}")
+                    val marketNote: MarketNote = response.body()
+                    Notebook.Note(marketNote)
+                }
+            }
+        } catch (_: Throwable) {
+            null
+        }
+    }
+
+    override suspend fun post(note: MarketNote): Boolean {
+        return try {
```

Comment:
nit: can make these functions more expressive by removing their brackets and return statements.